web-local/src/routes/(viz)/dashboard/[name]/+page.svelte

```diff
@@ -24,26 +25,45 @@
   $: ({ instanceId } = $runtime);
 
   $: filePaths = data.metricsView.meta?.filePaths as string[];
-  $: projectParserQuery = useProjectParser(queryClient, instanceId);
+  $: dashboard = useDashboard(instanceId, metricsViewName);
+  $: measures = $dashboard.data?.metricsView?.state?.validSpec?.measures ?? [];
+  $: projectParserQuery = useProjectParser(
+    queryClient,
+    instanceId,
+    measures.length > 0,
+  );
   $: dashboardFileHasParseError =
     $projectParserQuery.data?.projectParser?.state?.parseErrors?.filter(
       (error) => filePaths.includes(error.filePath as string),
     );
-
-  $: dashboard = useDashboard(instanceId, metricsViewName);
   $: mockUserHasNoAccess =
     $selectedMockUserStore && $dashboard.error?.response?.status === 404;
+
+  // Handle errors from dashboard YAML edits from an external IDE
+  $: if (dashboardFileHasParseError && dashboardFileHasParseError.length > 0) {
+    eventBus.emit("banner", {
+      type: "error",
+      message:
+        "Error parsing dashboard – you are viewing your last valid dashboard specification",
+      iconType: "alert",
+    });
+  }
```

Comment:
So I tried this out and, apologies, I'm observing different behavior from what I originally expected. Actually, a parse error results in the resource being lost outright. There's no valid spec retained. So it seems we should revert this and show the "Error parsing dashboard" error page.